devdoc/slist.md

```diff
@@ -8,164 +8,178 @@
 ## Exposed API
 
 ```c
-typedef struct SLIST_ENTRY_TAG
+typedef struct SINGLYLINKEDLIST_ENTRY_TAG
 {
-    struct SLIST_ENTRY_TAG *next;
-} SLIST_ENTRY, *PSLIST_ENTRY;
-
-typedef bool (*SLIST_MATCH_FUNCTION)(PSLIST_ENTRY list_entry, const void* match_context);
-typedef bool (*SLIST_CONDITION_FUNCTION)(PSLIST_ENTRY list_entry, const void* match_context, bool* continue_processing);
-typedef int (*SLIST_ACTION_FUNCTION)(PSLIST_ENTRY list_entry, const void* action_context, bool* continue_processing);
-
-MOCKABLE_FUNCTION(, bool, slist_initialize, PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, bool, slist_is_empty, const PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_add, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_add_head, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_remove, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_remove_head, PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_find, PSLIST_ENTRY, list_head, SLIST_MATCH_FUNCTION, match_function, void*, match_context);
-MOCKABLE_FUNCTION(, int, slist_remove_if, PSLIST_ENTRY, list_head, SLIST_CONDITION_FUNCTION, condition_function, void*, match_context);
-MOCKABLE_FUNCTION(, int, slist_for_each, PSLIST_ENTRY, list_head, SLIST_ACTION_FUNCTION, action_function, void*, action_context);
+    struct SINGLYLINKEDLIST_ENTRY_TAG *next;
+} SINGLYLINKEDLIST_ENTRY, *PSINGLYLINKEDLIST_ENTRY;
+
+typedef enum
+{
+    EMPTY,
```

Comment:
>EMPTY [](http://example.com/codeflow?start=4&length=5)

These have a very high chance to collide with other enum values. We typically use a prefix which is the enum name, like:

SLIST_IS_EMPTY_RESULT_EMPTY